### PR TITLE
Initial refactor: combining BMS_Read functions and implementing BMS_P…

### PR DESCRIPTION
**Pull request by amneetm**
…rocessState

### CODE REVIEW

File: BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h [hunk 1]
```diff
@@ -9,23 +9,48 @@
 
 #include "sys_common.h"
 
+typedef enum State_t
+{
+    BMS_RUNNING,
+    BMS_CHARGING,
+    BMS_FAULT
+}State_t;
```
Comment (mkamalel) [inline BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h]: Name of the struct is vague, have it specified that it's related to BMS

File: BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h [hunk 2]
```diff
@@ -9,23 +9,48 @@
...
+	uint8_t 3SECOND_FLAG; // If 1, then a cell has been in fault for longer than 3 seconds, else 0
+						  // TODO: If 1, put BMS in FAULT state
+	uint8_t TOTAL_CELL_ERROR_FLAG; // If 1, then more than four cells have been in fault during a check, else 0
+								   // TODO: If 1, put BMS in FAULT state
+	uint8_t BAD_SLAVE_CONNECTION_FLAG; // If 1, then BMS master is unable to read/write to/from at least one BMS slave
+									   // TODO: If 1, put BMS in FAULT state
 } flags;
 
-typedef struct SOCInfo
+typedef struct data
```
Comment (mkamalel) [inline BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h]: Same as above comment, you can keep it as data but define it as a static variable so its not accessible by other files

File: BMS Workspace-git/bms-1227-fw/Phantom_drivers/soc/soc.c [hunk 3]
```diff
@@ -31,7 +31,7 @@ float battCapacity = 0; // Max chemical capacity for cell TODO: Make static afte
 static TickType_t xLastWakeTime = 0;
 float lastCurrentValue = 0; // TODO: Make static after unit-testing
 
-State_t state = CHARGE; // TODO: Make static after unit-testing 
+SOCState_t SOCState = CHARGE; // TODO: Make static after unit-testing 
 bms_data* BMSDataPtr; // TODO: Make this extern and initialize BMSDataPtr when BMS is initialized
```
Comment (mkamalel) [inline BMS Workspace-git/bms-1227-fw/Phantom_drivers/soc/soc.c:35]: I'd rather you have the pointer as an input to the init function and set it that way

Comment (amneetm) [reply BMS Workspace-git/bms-1227-fw/Phantom_drivers/soc/soc.c:35]: CHARGE was the wrong thing to initialize it to. I actually mean to initialize it to INIT, and then the state machine knows to call the SOCinit() function based on this state. Does that flow make sense?

Comment (amneetm) [reply BMS Workspace-git/bms-1227-fw/Phantom_drivers/soc/soc.c:35]: Changed to be initialized to INIT.
This way, the state machine knows to call the SOCinit() function based on this state the first time SOCprocess() is called. Does that flow make sense?

File: BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h (hunk 2 above)
Comment (amneetm) [reply BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h]: Changed all member variable types to start with "bms" (ex: data -> bmsData)

File: BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h (hunk 1 above)
Comment (amneetm) [reply BMS Workspace-git/bms-1227-fw/Phantom_drivers/include/bms_data.h]: Changed to BMSState_t

**Review by mkamalel — APPROVED** [review]

**Pull request merged**